client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1287 @@
+				eventOptions
+			);
+			recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_main_account', eventOptions );
+		}
+
+		if ( ! prevShowEmailCodeVerification.current && showEmailCodeVerification ) {
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_code_verification',
+				eventOptions
+			);
+		}
+
+		if ( ! prevShowCheckYourEmail.current && showCheckYourEmail ) {
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_link_verification',
+				eventOptions
+			);
+		}
+
+		// Update refs for next render
+		prevShowCheckYourEmail.current = showCheckYourEmail;
+		prevShowEmailCodeVerification.current = showEmailCodeVerification;
+		prevShowSecondaryEmailOptions.current = showSecondaryEmailOptions;
+	}, [
+		showCheckYourEmail,
+		showEmailCodeVerification,
+		showSecondaryEmailOptions,
+		oauth2ClientId,
+		oauth2Client?.title,
+		isGravatarFlow,
+		currentQueryArguments?.gravatar_from,
+		currentQueryArguments?.email_address,
+	] );
+
+	const resendEmailCountdownId = useRef< NodeJS.Timeout | null >( null );
+	const [ resendEmailCountdown, setResendEmailCountdown ] = useState( RESEND_EMAIL_COUNTDOWN_TIME );
+
+	const resetResendEmailCountdown = () => {
+		if ( resendEmailCountdownId.current ) {
+			clearInterval( resendEmailCountdownId.current );
+			resendEmailCountdownId.current = null;
+		}
+
+		setResendEmailCountdown( RESEND_EMAIL_COUNTDOWN_TIME );
+	};
+
+	const startResendEmailCountdown = () => {
+		resetResendEmailCountdown();
+
+		resendEmailCountdownId.current = setInterval( () => {
+			setResendEmailCountdown( ( prev ) => {
+				if ( prev <= 1 ) {
+					if ( resendEmailCountdownId.current ) {
+						clearInterval( resendEmailCountdownId.current );
+						resendEmailCountdownId.current = null;
+					}
+					return 0;
+				}
+				return prev - 1;
+			} );
+		}, 1000 );
+	};
+
+	// Ensure interval is cleared on unmount
+	useEffect( () => {
+		return () => {
+			if ( resendEmailCountdownId.current ) {
+				clearInterval( resendEmailCountdownId.current );
+				resendEmailCountdownId.current = null;
+			}
+		};
+	}, [] );
+
+	const handleGravPoweredEmailCodeSend = async ( email: string, cb = () => {} ) => {
+		const noticeId = 'email-code-notice';
+		const duration = 4000;
+		const eventOptions = { client_id: oauth2ClientId, client_name: oauth2Client?.title };
+
+		setIsRequestingEmail( true );
+
+		try {
+			infoNotice( translate( 'Sending email…' ), { id: noticeId, duration } );
+
+			recordTracksEvent(
+				'calypso_gravatar_powered_magic_login_email_code_requesting',
+				eventOptions
+			);
+
+			const { public_token } = await wpcom.req.post(
+				'/auth/send-login-email',
+				{ apiVersion: '1.3' },
+				{
+					client_id: config( 'wpcom_signup_id' ),
+					client_secret: config( 'wpcom_signup_key' ),
+					locale,
+					lang_id: getLanguage( locale )?.value,
+					email: email,
+					redirect_to: currentQueryArguments?.email_address,
```

Comment:
@chriskmnds This value should be the `redirect_to`, or we will get the wrong logged-in direction:

```suggestion
					redirect_to: currentQueryArguments?.redirect_to,
```